org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMAttributeFormatter.java

```diff
@@ -78,7 +73,33 @@ public void formatAttribute(DOMAttr attr, int prevOffset, boolean singleAttribut
 				int attrValueStart = attr.getNodeAttrValue().getStart(); // <foo attr = |""
 				removeLeftSpaces(delimiterOffset, attrValueStart, edits);
 			}
-			formatAttributeValue(attr, parentConstraints.getIndentLevel(), edits);
+			// Add width for length of attribute name and 3 for '=""'
+			// between start tag name and tag
+			attributeNamelength = attrNameEnd - attr.getNodeAttrName().getStart() + 3;
+			parentConstraints.setAvailableLineWidth(availableLineWidth - attributeNamelength);
+			formatAttributeValue(attr, parentConstraints, edits);
+		}
+
+		if (!alreadyIndented) {
+			int from = prevOffset;
+			int to = attr.getStart();
+			availableLineWidth = parentConstraints.getAvailableLineWidth();
+			if (availableLineWidth < 0 && !isSplitAttributes()
+					&& !XSISchemaModel.isXSISchemaLocationAttr(attr.getNodeName(), attr)) {

```

Comment:
XSISchemaModel is an extension, so you should not use XSISchemaModel here. 

I wonder why you need to do a test with XSISchemaModel?

If you need really do this test, do that with IFormatterParticipant by adding it a new method.